src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.Analyzer.cs

```diff
@@ -17,9 +18,62 @@ private Analyzer()
             {
             }
 
-            public static SyntaxKind Analyze(SwitchStatementSyntax node, out bool shouldRemoveNextStatement)
+            public static (SyntaxKind nodeToGenerate, VariableDeclaratorSyntax declaratorToRemoveOpt) Analyze(
+                SwitchStatementSyntax node,
+                SemanticModel semanticModel,
+                out bool shouldRemoveNextStatement)
             {
-                return new Analyzer().AnalyzeSwitchStatement(node, out shouldRemoveNextStatement);
+                var analyzer = new Analyzer();
+                var nodeToGenerate = analyzer.AnalyzeSwitchStatement(node, out shouldRemoveNextStatement);
+
+                if (nodeToGenerate == SyntaxKind.SimpleAssignmentExpression &&
+                    analyzer.TryGetVariableDeclaratorAndSymbol(semanticModel) is var (declarator, symbol))
+                {
+                    if (shouldRemoveNextStatement &&
+                        semanticModel.AnalyzeDataFlow(node.GetNextStatement()).DataFlowsIn.Contains(symbol))
+                    {
+                        // Bail out if data flows into the next statement that we want to move
```

Comment:
Instead of bailing, can we just not remove the next statement and leave it outside the switch?

Reply:
Just looking through the `AnalyzeSwitchStatement` implementation, we're setting `shouldRemoveNextStatement` always to true if there is no default. That's potentially wrong, especially in the case noted in #37949 

Maybe we can check if there is a default that makes sense before assuming we need to remove next?

Reply:
> Instead of bailing, can we just not remove the next statement and leave it outside the switch?

At that point, we must have a viable statement after switch, otherwise this is not an exhaustive switch and we can't offer the fix -- `AnalyzeNextStatement` returns `default`